src/back-end/lib/resources/proposal/code-with-us.ts

```diff
@@ -0,0 +1,403 @@
+            if (isInvalid(validatedWithdrawalNote)) {
+              return invalid({ proposal: adt('withdraw' as const, validatedWithdrawalNote.value) });
+            }
+            return valid({
+              session: request.session,
+              body: adt('withdraw', validatedWithdrawalNote.value)
+            });
+          default:
+            return invalid({ proposal: adt('parseFailure' as const) });
+        }
+      },
+      respond: wrapRespond<ValidatedUpdateRequestBody, UpdateValidationErrors, JsonResponseBody<CWUProposal>, JsonResponseBody<UpdateValidationErrors>, Session>({
+        valid: (async request => {
+          let dbResult: Validation<CWUProposal, null>;
+          const { session, body } = request.body;
+          switch (body.tag) {
+            case 'edit':
+              dbResult = await db.updateCWUProposal(connection, { ...body.value, id: request.params.id }, session);
+              break;
+            case 'submit':
+              dbResult = await db.updateCWUProposalStatus(connection, request.params.id, CWUProposalStatus.Submitted, body.value, session);
+              break;
+            case 'score':
+              dbResult = await db.updateCWUProposalScore(connection, request.params.id, body.value.score, body.value.note, session);
+              break;
+            case 'award':
+              dbResult = await db.awardCWUProposal(connection, request.params.id, body.value, session);
+              break;
+            case 'disqualify':
+              dbResult = await db.updateCWUProposalStatus(connection, request.params.id, CWUProposalStatus.Disqualified, body.value, session);
+              break;
+            case 'withdraw':
+              dbResult = await db.updateCWUProposalStatus(connection, request.params.id, CWUProposalStatus.Withdrawn, body.value, session);
+              break;
+          }
+          if (isInvalid(dbResult)) {
+            return basicResponse(503, request.session, makeJsonResponseBody({ database: [db.ERROR_MESSAGE] }));
+          }
+          return basicResponse(200, request.session, makeJsonResponseBody(dbResult.value));
+        }),
+        invalid: (async request => {
+          return basicResponse(400, request.session, makeJsonResponseBody(request.body));
+        })
+      })
+    };
+  },
+
+  delete(connection) {
+    return {
+      async validateRequestBody(request) {
+        if (!(await permissions.deleteCWUProposal(connection, request.session, request.params.id))) {
+          return invalid({
+            permissions: [permissions.ERROR_MESSAGE]
+          });
+        }
+        const validatedCWUProposal = await validateCWUProposalId(connection, request.params.id, request.session);
+        if (isInvalid(validatedCWUProposal)) {
+          return invalid({ notFound: ['Proposal not found.'] });
+        }
+        if (validatedCWUProposal.value.status !== CWUProposalStatus.Draft) {
+          return invalid({ permissions: [permissions.ERROR_MESSAGE] });
```

Comment:
Change this error message:

`{ status: ['You can not delete a proposal that is not a draft.'] }`